### ISSUE TITLE
Feature request: add CSV vtable module

### ISSUE THREAD

**Issue by kenshaw** [mattn/go-sqlite3#393]
The SQLite3 source provides a [CSV vtable module](https://www.sqlite.org/csv.html), but it is not part of the amalgamated source. If possible, it would be great if the csv vtable module could be included in the base `go-sqlite3` package, along with a related `csv` build tag. 

This would effectively allow the following SQL:

```sql
CREATE VIRTUAL TABLE temp.t1 USING csv(filename='thefile.csv');

SELECT * FROM temp.t1;
```

Rationale: 

It seems disjoint that there are other modules (such as `json1`) that are part of the amalgamated source, but that require separate build tags to enable. I understand this is likely due to a want to keep the build times as low as possible, and thus disabling extraneous features, but for completeness sake, the other base/core modules (those that have `.html` documentation) should at least be available via build tags.

I am happy to create a separate repository and connect up the CSV module that way, but considering how widely CSV is used, it would make sense for SQLite's core/original CSV be packaged with this repository. As such, I would be more than willing to do the work to get the csv module added to the code, and submit a proper pull request, however I would like to get a "thumbs up" before embarking on any work. If including the csv module directly is out of the question, I will just create a separate repository and connect using the appropriate hooks.

(note: I am working on some advanced features for github.com/knq/usql and would like CSV to be natively available for sqlite databases)

(btw, @mattn thanks for all your amazing Go code/repositories....!)

**Comment by ekzhu** [comment]
Agreed. It would be very helpful to have CSV module.

**Comment by mattn** [comment]
Thanks for your suggestion. I totally agree with you. As you know, CSV module is separated from codes of core. To enable CSV module, you must compile csv.so with getting code from https://www.sqlite.org/src/artifact?ci=trunk&filename=ext/misc/csv.c . And load the module like below.

```go
	sql.Register("sqlite3_with_extensions",
		&sqlite3.SQLiteDriver{
			Extensions: []string{
				"csv",
			},
		})
```

However, I often feel trouble to use go-sqlite3 for unit test. For example, we want to use go-sqlite3 to load test data from CSV, as you mentioned. So I agree with you in personally.

**Comment by patrickdevivo** [comment]
Hi there, I just wanted to ask about this as it was closed by a commit (c9394b1e02281de8e1d60049383d710005cc0ae5), but I can't seem to find any implementation of it. Is it necessary to load the csv virtual table as an extension still, rather than via a build tag? It would be great if it were supported similar to other extensions enumerated in the extensions list
